pydantic/_internal/_std_types_schema.py

```diff
@@ -307,141 +294,296 @@ def validate(self, __input_value: int | float | str) -> Decimal:  # noqa: C901 (
-        metadata=metadata,
     )
 
-    return core_schema.lax_or_strict_schema(
+    schema = core_schema.lax_or_strict_schema(
         lax_schema=lax,
         strict_schema=strict,
         serialization=core_schema.to_string_ser_schema(),
     )
 
+    return [
+        source_type,
+        InnerSchemaValidator(schema, js_core_schema=core_schema.str_schema(), js_schema_update={'format': 'uuid'}),
+        *annotations,
+    ]
+
 
-@schema_function(PurePath)
-@schema_function(PathLike)
-def path_schema(_schema_generator: GenerateSchema, path_type: type[PathLike[Any]]) -> core_schema.LaxOrStrictSchema:
-    construct_path = PurePath if path_type is PathLike else path_type
-    metadata = build_metadata_dict(js_functions=[lambda _c, _h: {'type': 'string', 'format': 'path'}])
+def path_schema_prepare_pydantic_annotations(source_type: Any, annotations: Iterable[Any]) -> list[Any] | None:
+    import pathlib
 
-    def path_validator(input_value: str) -> PathLike[Any]:
+    if source_type not in {
+        os.PathLike,
+        pathlib.Path,
+        pathlib.PurePath,
+        pathlib.PosixPath,
+        pathlib.PurePosixPath,
+        pathlib.PureWindowsPath,
+    }:
+        return None
+
+    metadata, remaining_annotations = _known_annotated_metadata.collect_known_metadata(annotations)
+    _known_annotated_metadata.check_metadata(metadata, _known_annotated_metadata.STR_CONSTRAINTS, source_type)
+
+    construct_path = pathlib.PurePath if source_type is os.PathLike else source_type
+
+    def path_validator(input_value: str) -> os.PathLike[Any]:
         try:
             return construct_path(input_value)  # type: ignore
         except TypeError as e:
             raise PydanticCustomError('path_type', 'Input is not a valid path') from e
 
+    constrained_str_schema = core_schema.str_schema(**metadata)
+
     instance_schema = core_schema.json_or_python_schema(
-        json_schema=core_schema.no_info_after_validator_function(path_validator, core_schema.str_schema()),
-        python_schema=core_schema.is_instance_schema(path_type),
+        json_schema=core_schema.no_info_after_validator_function(path_validator, constrained_str_schema),
+        python_schema=core_schema.is_instance_schema(source_type),
     )
 
-    return core_schema.lax_or_strict_schema(
+    schema = core_schema.lax_or_strict_schema(
         lax_schema=core_schema.union_schema(
             [
                 instance_schema,
-                core_schema.no_info_after_validator_function(path_validator, core_schema.str_schema()),
+                core_schema.no_info_after_validator_function(path_validator, constrained_str_schema),
             ],
-            custom_error_type='path_type',
-            custom_error_message='Input is not a valid path',
-            strict=True,
+            # custom_error_type='path_type',
+            # custom_error_message='Input is not a valid path',
+            # strict=True,
```

Comment:
should these commented lines be removed?

Reply:
Oh yes good catch